presto-hive/src/main/java/io/prestosql/plugin/hive/metastore/thrift/ThriftHiveMetastore.java

```diff
@@ -1333,6 +1318,48 @@ private boolean containsAllPrivilege(Set<PrivilegeGrantInfo> requestedPrivileges
                 .anyMatch(privilege -> privilege.getPrivilege().equalsIgnoreCase("all"));
     }
 
+    @SafeVarargs
+    private final <T> T alternativeCall(
+            AtomicInteger verifiedAlternatives,
+            Call<T>... alternatives)
+            throws TException
+    {
+        checkArgument(alternatives.length > 0, "No alternatives");
+
+        int state = verifiedAlternatives.get();
+        if (state != 0) {
+            for (int position = 0; position < alternatives.length; position++) {
+                if ((state & (1 << position)) != 0) {
+                    try (ThriftMetastoreClient client = createMetastoreClient()) {
+                        return alternatives[position].callOn(client);
+                    }
+                }
+            }
+        }
+
+        Exception firstException = null;
+        for (int i = 0; i < alternatives.length; i++) {
+            int position = i;
+            try (ThriftMetastoreClient client = createMetastoreClient()) {
+                T result = alternatives[i].callOn(client);
+                verifiedAlternatives.updateAndGet(currentState -> currentState | (1 << position));
+                return result;
```

Comment:
you are using `verifiedAlternatives` as a bit mask, which suggests that there could be many alternatives enabled. However from what I understand, there could be only single one be enabled.  Then bit mask can be refactored to an regular alternative method index.

Reply:
We can concurrently realize two alternatives work, so technically there can be more than one bit set.
Still... I will simplify.